server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -74,15 +74,16 @@ def validate
     end
 
     def log
-      reader = Fusor::Logging::LogReader.new
-      log_path = ::Fusor.log_file_path(@deployment.name, @deployment.id)
+      log_type_param = params[:log_type] || 'fusor_log'
+      reader = create_log_reader(log_type_param)
```

Comment:
@daviddavis In order to test out this function, it seems like I'd need to stub out create_log_reader to return mock LogReaders.  What's the preferred way to do this?  I'll address the testing of this controller in a separate PR.